docs/user_needs/software_developer.sdoc

```diff
@@ -0,0 +1,23 @@
+[DOCUMENT]
+TITLE: Software Developer
```

Comment:
To justify the creation of a separate fragment for Software Developer needs, please come up with at least three compelling examples of developer needs.

Do we have at least one real software developer need in the existing documents?

(I fully support the idea of extracting the user stories as User Needs but I cannot think of enough developer needs collected in the existing documents so far.)

Reply:
Personally, I don't like "Zephyr RTOS User" and "Software Developer" as terms. These needs exist outside of the consideration of Zephyr Project. I think the following highest-level categories fit better. Within each there are multiple roles and skill levels which have varying needs. I've included some examples of real existing-but-still-to-be-captured needs for each:

- Project Users (e.g., Product Manufacturers, Chip Vendors, CS/SwEng/CompEng/EE Students)
  - Includes a guide to getting started with the ecosystem;
  - Able to develop on Windows, MacOS, or Linux environments;
  - Able to submit issues to the project;
  - Licensed such that it can be freely integrated into a product without having to publish proprietary content.
- Project Contributors (e.g., Product Manufacturers, Chip Vendors, Contractors)
  - Includes a guide to contributing to the project
  - Able to submit PRs to issues;
  - Able to develop in a Docker container so that it matches the CI pipeline environment;
- Stakeholders (e.g., Regulators, Certification Test Houses, Linux Foundation, Platinum Members; Product Manufacturer Sponsors)
  - Able to restrict licensing of contributions and released content;
  - Enforce certifiable content development follows an approved SDLC;
  - Project Certifiable Release is extensible to enable reuse on proprietary HW.